src/background/details-view-controller.ts

```diff
@@ -18,20 +18,19 @@ export class DetailsViewController {
         this.detailsViewRemovedHandler = handler;
     }
 
-    public showDetailsView(targetTabId: number): void {
+    public showDetailsView(targetTabId: number): Promise<void> {
         const detailsViewTabId = this.tabIdToDetailsViewMap[targetTabId];
 
         if (detailsViewTabId != null) {
             this.browserAdapter.switchToTab(detailsViewTabId);
             return;
         }
 
-        this.browserAdapter.createTabInNewWindow(
-            this.getDetailsUrl(targetTabId),
-            (tab: chrome.tabs.Tab) => {
+        return this.browserAdapter
```

Comment:
If you use async/await in this sort of scenario, you'll get more useful error stacks in the event that something goes wrong (the stack will include this function with await, but will omit it without)

Reply:
@dbjorge updated